course_access_groups/models.py

```diff
@@ -52,6 +56,36 @@ class MembershipRule(utils_models.TimeStampedModel):
     domain = models.CharField(max_length=255, db_index=True, help_text='The learner email domain e.g. "example.com".')
     group = models.ForeignKey(CourseAccessGroup, on_delete=models.CASCADE)
 
+    @classmethod
+    def apply_for_user(cls, user):
+        """
+        Automatically enroll a user based on existing MembershipRule.
+        :param user:
+        :return: Membership
+        """
+        assert user.is_active, 'Ensure that only users with verified emails are enrolled the group'
```

Comment:
Do you plan on keeping this assert in production?

Reply:
Maybe not. I've refactored it to `ValueError` and added a longer comment to explain why this should not happen in production anyway.